src/flux/course-listing.coffee

```diff
@@ -70,4 +70,9 @@ CourseListingStore = flux.createStore
     allCourses: ->
       return _.compact _.map @_course_ids, CourseStore.get
 
+    allCoursesWithRoles: ->
+      return _.compact _.map @_course_ids, (id) ->
+        course = CourseStore.get(id)
+        course if course?.roles?.length
+
```

Comment:
A stylistic nitpick, but this condition could be `_.isEmpty(CourseStore.get(id)?.roles)` since isEmpty tolerates nulls, which could turn this into:
```
   allCoursesWithRoles: ->
      _.compact _.map @_course_ids, (id) -> !_.isEmpty( CourseStore.get(id)?.roles )
```